### PR TITLE
Add `SKNode` anchor points

### PR DESCRIPTION
**Pull request by rypyak**
Add `SKNode` anchor points for `center`, `topLeft`, `topRight`, `bottomLeft` and `bottomRight`

* Add Unit-Tests for each anchor point get and set operation
* Change `SpriteKitTests` file and enclosing class names to `SKNodeExtensionTest`

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by rypyak** [comment]
Wasn't really sure how to get the `#PR` number to add into `CHANGELOG.md` before actually opening it, can you enlighten me please?

**Comment by LucianoPAlmeida** [comment]
> Wasn't really sure how to get the `#PR` number to add into `CHANGELOG.md` before actually opening it, can you enlighten me please?

Normally changelog is an extra commit after the PR is opened :)

File: Tests/SpriteKitTests/SKNodeExtensionTests.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+//  Copyright © 2018 SwifterSwift
+//
+
+import XCTest
+@testable import SwifterSwift
+
+#if canImport(SpriteKit)
+import SpriteKit
+
+@testable import SwifterSwift
```
Comment (guykogus) [inline Tests/SpriteKitTests/SKNodeExtensionTests.swift:15]: You can remove this.

File: Tests/SpriteKitTests/SKNodeExtensionTests.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        scene.addChild(childOne)
+        let childTwo = SKNode()
+        childOne.addChild(childTwo)
+        XCTAssertEqual(scene.descendants(), [childOne, childTwo])
+        XCTAssertEqual(childOne.descendants(), [childTwo])
+        XCTAssertEqual(childTwo.descendants(), [])
+    }
+
+    func testSKNodeCenter() {
+        let scene = SKScene(size: .init(width: 100, height: 100))
```
Comment (guykogus) [inline Tests/SpriteKitTests/SKNodeExtensionTests.swift:31]: The standard in this project (although I'm sure it's not used in a few places) is to use the explicit initialiser.
```suggestion
        let scene = SKScene(size: CGSize(width: 100, height: 100))
```

File: Tests/SpriteKitTests/SKNodeExtensionTests.swift [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    func testTopLeft() {
+        let scene = SKScene(size: .init(width: 100, height: 100))
+        let nodeSize = CGSize(width: 50, height: 50)
+        let node = SKSpriteNode(color: .black, size: nodeSize)
+        scene.addChild(node)
+
+        node.topLeft = CGPoint(x: scene.frame.midX, y: scene.frame.midY)
+        XCTAssertEqual(node.topLeft, CGPoint(x: 50, y: 50))
+        XCTAssertEqual(node.frame.size, nodeSize)
+        XCTAssertEqual(node.frame.origin, CGPoint(x: 50, y: 00))
```
Comment (guykogus) [inline Tests/SpriteKitTests/SKNodeExtensionTests.swift:51]: One zero is enough. (Fix all instances, please.)
```suggestion
        XCTAssertEqual(node.frame.origin, CGPoint(x: 50, y: 0))
```

File: Tests/SpriteKitTests/SKNodeExtensionTests.swift (hunk 1 above)
Comment (rypyak) [reply Tests/SpriteKitTests/SKNodeExtensionTests.swift:15]: Good spot, I've got no idea why it would be there in the original file. 
🧹 💨

**Comment by LucianoPAlmeida** [comment]
@rypyak Can you also resolve the conflicts in changelog? :)

File: Tests/SpriteKitTests/SKNodeExtensionTests.swift (hunk 2 above)
Comment (rypyak) [reply Tests/SpriteKitTests/SKNodeExtensionTests.swift]: I've added review changes and squashed all into a single commit for convenience. 🚀 
Please re-review @LucianoPAlmeida @guykogus

**Comment by rypyak** [comment]
I've added review changes and squashed all into a single commit for convenience. 🚀
Please re-review @LucianoPAlmeida @guykogus

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Pull request merged**